### DISCUSSION TITLE
How many uart lgt8fx?

### DISCUSSION THREAD

**Discussion by jokokiswan** [dbuezas/lgt8fx#252] [Q&A] [answered]
Got rx0 tx0 and tx1 rx1, how to access on diferent physical uart if got 2 uart?

Thank you.

**Comment by LaZsolt** [comment]
As far as I know, there no second hardware uart in LGT8Fx,

Where did you read about  rx0 tx0 and tx1 rx1 are in LGT8Fx?

**Reply by jokokiswan** [reply]
![images (1)](https://user-images.githubusercontent.com/58895576/222879134-b79fcdb8-f8c7-457a-a3fd-6de5f260521a.png)
Am i wrong ? TxD not tx0?

**Reply by LaZsolt** [reply]
Yes TxD is TX0.
But no Tx1 or Rx1 on the picture.

**Comment by jayzakk** [comment]
@jokokiswan , you may have seen the alternate pins for Rx/tx. You can switch the function to different pins.

Basically you may be able to use spi as uart, but afaik no such library till now.

**Reply by jokokiswan** [reply]
Ok, i m a beginner :), it seems very hard to do.

**Comment by dbuezas** [comment, marked as answer]
What you can do is use the SoftwareSerial library. Here's an [example](https://github.com/dbuezas/lgt8fx/blob/master/lgt8f/libraries/SoftwareSerial/examples/SoftwareSerialExample/SoftwareSerialExample.ino)

**Reply by jokokiswan** [reply]
Ok, thank you

**Reply by dbuezas** [reply]
Wish you luck!
The software serial library is very flexible (multiple serials, you can pick which pins, etc), but it doesn't work well at high baud rates and is heavier on the cpu. So make sure to use lower baud rates when you can.

**Reply by jokokiswan** [reply]
Ok bro, thank you.
